.github/actions/with-docker/action.yml

```diff
@@ -37,3 +37,4 @@ runs:
 
       docker cp . ${TAG_NAME}:/home/user
       docker exec ${TAG_NAME} chown -R user:user /home/user
+      docker exec -u user ${TAG_NAME} cat /home/user/deps/plugin/deps/libff/build/CMakeCache.txt
```

Comment:
This demonstrates why the following error occurs ([build](https://github.com/runtimeverification/evm-semantics/actions/runs/3693720518/jobs/6254042903#step:5:40)):

```
CMake Error: The current CMakeCache.txt directory /home/user/deps/plugin/deps/libff/build/CMakeCache.txt is different than the directory /opt/workspace/deps/plugin/deps/libff/build where CMakeCache.txt was created. This may result in binaries being created in the wrong place. If you are not sure, reedit the CMakeCache.txt
```

Output ([build](https://github.com/runtimeverification/evm-semantics/actions/runs/3693720518/jobs/6254042903#step:3:171)):
```
docker exec -u user kevm-ci-concrete-7a392c544137dd6fb49dd8e3b1bb08d02d9f6e74 cat /home/user/deps/plugin/deps/libff/build/CMakeCache.txt
# This is the CMakeCache file.
# For build in directory: /opt/workspace/deps/plugin/deps/libff/build
...
```

So currently `/opt/workspace` contains checked out code that has already been built (at least in part). The `checkout` action should ensure the repo is cleaned, but the `clean` parameter only seems to run `git clean -ffdx` without recursing into submodules.

The PR can thus be unblocked by ensuring that the repo copied into the docker container is in a clean state, probably by doing some cleanup on the runner.

Reply:
Clean is not acting as it should. According to docs from git-clean -d should enable recursive cleaning 
```
-d
           Normally, when no <path> is specified, git clean will not recurse into untracked directories to avoid removing too
           much. Specify -d to have it recurse into such directories as well. If any paths are specified, -d is irrelevant; all
           untracked files matching the specified paths (with exceptions for nested git directories mentioned under --force) will
           be removed.
```